browser/ui/webui/settings/brave_site_settings_handler.cc

```diff
@@ -0,0 +1,69 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/webui/settings/brave_site_settings_handler.h"
+
+#include <string>
+
+#include "base/values.h"
+#include "brave/components/brave_shields/browser/brave_shields_util.h"
+#include "chrome/browser/ui/webui/settings/site_settings_helper.h"
+#include "components/grit/brave_components_strings.h"
+#include "ui/base/l10n/l10n_util.h"
+
+namespace settings {
+
+namespace {
+constexpr char kIsValidKey[] = "isValid";
+constexpr char kReasonKey[] = "reason";
+}  // namespace
+
+BraveSiteSettingsHandler::BraveSiteSettingsHandler(Profile* profile)
+    : SiteSettingsHandler(profile) {}
+
+BraveSiteSettingsHandler::~BraveSiteSettingsHandler() = default;
+
+void BraveSiteSettingsHandler::RegisterMessages() {
+  web_ui()->RegisterMessageCallback(
+      "isPatternValidForType",
+      base::BindRepeating(
+          &BraveSiteSettingsHandler::HandleIsPatternValidForType,
+          base::Unretained(this)));
+
+  SiteSettingsHandler::RegisterMessages();
+}
+
+bool BraveSiteSettingsHandler::IsPatternValidForBraveContentType(
+    ContentSettingsType content_type,
+    const std::string& pattern_string) {
+  if (content_type != ContentSettingsType::BRAVE_SHIELDS)
+    return true;
+  return (pattern_string.find('*') == std::string::npos);
+}
+
+void BraveSiteSettingsHandler::HandleIsPatternValidForType(
+    const base::Value::List& args) {
+  CHECK_EQ(3U, args.size());
```

Comment:
looks like you're missing an `AllowJavascript();` call here

Reply:
it is on line 61